### DISCUSSION TITLE
DR questions..

### DISCUSSION THREAD

**Discussion by ts97783** [backube/scribe#135] [Q&A]
Hi Guys, 

I'm very interested in using Scribe for Disaster Recovery between two on-premise OpenShift Clusters.

How are you replicating OpenShift Cluster specific config to a secondary site (Projects, NameSpaces, Roles etc)?

I appreciate Scribe is managing the Persistent Volume replication, I'm curious on how you guys replicate all the other OpenShift configuration into a DR cluster.

Thanks, 

Tim

**Comment by JohnStrunk** [comment]
Our current recommendation is to use a gitops-like setup. The yamls that describe your app would reside in a git repo, and they would be applied to the two clusters either automatically via something like [Argo CD](https://argoproj.github.io/argo-cd/) or manually via [kustomize](https://kustomize.io/).

In the case of kustomize, you would have a common base description of your app that has the parts that should be identical on both clusters. You would then have "patches" that sit alongside the base description, adjusting the configuration for site or role-specific items.

For example, in some of the demos I've done, the PVC for the app is customized between the two sites.
- A "primary" role with the PVC as normal
- A "scaled down secondary" that has all the components, but Deployments scaled to 0 and no PVC
- A "scaled up secondary" that patches the Deployments back to 1 (or whatever), plus the PVC description with a volumesource that creates it from the Scribe snapshot.

You may not want/need the scaled down secondary configuration at all. I only used it so that the AWS ELB was immediately available when I switch over.

**Reply by ts97783** [reply]
Thanks very much for the reply. This is exactly the information I needed.
